lib/iterator.js

```diff
@@ -156,10 +156,30 @@ class Iterator {
     return new FlatMapIterator(this, mapper, thisArg);
   }
 
+  // Creates an iterator that aggregates elements from each of the iterators.
+  //
+  // Returns an iterator of Arrays where the i-th tuple contains
+  // the i-th element from each of the passed iterators.
+  // The iterator stops when the shortest input iterable is exhausted.
+  // Signature: ...iterators
+  //   iterators <Array>
+  // Returns: <Iterator>
   zip(...iterators) {
     return new ZipIterator(this, iterators);
   }
 
+  // Creates an iterator that aggregates elements from each of the iterators.
+  //
+  // If the iterables are of uneven length, missing values are filled-in
+  // with <undefined>. Iteration continues until the longest iterable
+  // is exhausted.
+  // Signature: ...iterators
+  //   iterators <Array>
+  // Returns: <Iterator>
+  zipLongest(...iterators) {
```

Comment:
I think we should allow specifying the value that should be used to fill with when there are no values left in one of the iterators.

Reply:
Yeah, I couldn't devise a way to make a proper API.